packages/storage-plus/src/snapshot/map.rs

```diff
@@ -205,6 +228,28 @@ where
         self.no_prefix_de().range_de(store, min, max, order)
     }
 
+    pub fn keys_de<'c>(
+        &self,
+        store: &'c dyn Storage,
+        min: Option<Bound>,
+        max: Option<Bound>,
+        order: cosmwasm_std::Order,
+    ) -> Box<dyn Iterator<Item = StdResult<K::Output>> + 'c>
+    where
+        T: 'c,
+        K::Output: 'static,
+    {
+        self.no_prefix_de().keys_de(store, min, max, order)
+    }
+
+    pub fn prefix_de(&self, p: K::Prefix) -> Prefix<K::Suffix, T> {
+        Prefix::new(self.primary.namespace(), &p.prefix())
+    }
+
+    pub fn sub_prefix_de(&self, p: K::SubPrefix) -> Prefix<K::SuperSuffix, T> {
```

Comment:
It is still a bit funny for me that the types here (connection between `p: K::SubPrefix` and `Prefix<K::SuperSuffix` seems to only be when writing this code and not enforced in the compile, but it is fine. Here we hide this chance to error from any user, and we are quite careful with this library code.

Mainly a thought for a future type-system

Reply:
Yes, we can easily mess up here. A type system with something like composite types, whose parts or elements could be referred to independently would be nice, yes.